src/library/scala/Option.scala

```diff
@@ -402,11 +406,16 @@ sealed abstract class Option[+A] extends Product with Serializable {
  *  @since   1.0
  */
 @SerialVersionUID(1234815782226070388L) // value computed by serialver for 2.11.2, annotation added in 2.11.4
-final case class Some[+A](value: A) extends Option[A] {
+final case class Some[+A] private[scala] (value: A) extends Option[A] {
   def isEmpty = false
   def get = value
 }
 
+object Some{
+  @deprecated("Create Some(...) directly is deprecated, use Option.apply(...) instead.")
```

Comment:
I'm not convinced of the value of deprecating this. Currently, best practice (at least as I understand it) is to use `Some(foo)` when you know `foo` will not be `null`, and `Option(foo)` when you think `foo` might be `null`. Put slightly differently, `Some(foo)` means "wrap the value `foo` in an `Option`", and `Option(foo)` means "only wrap the value in an `Option` if it is not `null`; if it is `null`, return an empty `Option`."

With this change, you will no longer be able to distinguish such uses, which makes the meaning of of `Option(foo)` less clear. Additionally, because this change supposes `Some(null)` is wrong, it also seems to suppose that `Option` is a better way of dealing with `null`, ***which it is not***. As explained quite well by @sjrd ([source](https://contributors.scala-lang.org/t/sip-suggestion-add-and-syntactic-sugar-for-more-convenient-option-t-usage/2413/9?u=nthportal)):

> **`Option` is not a better way to deal with `null`! It is a better way to deal with optional values.** `null` itself is a poor way to deal with optional values. It is not type parametric, as you cannot represent an optional value of a type that can itself hold `null`.

For this reason, I consider `Option.apply` to actually be a very poor default way of creating `Option`s. It equates `null` to `None` - which is incorrect - thus confusing the meaning of `Option` and muddying the meaning of code that uses it.

Reply:
As the Java `Optional`, it introduced the `Optional.of` and `Optional.ofNullalbe`